### ISSUE TITLE
Prepare the framework so that the tournament can be run on a VM

### ISSUE THREAD

**Issue by katjabercic** [KWARC/Kalah-Framework#6]
TODO:

- [ ] Can adding submissions into the project be improved?
- [ ] Log everything, including exceptions.
- [ ] Separate out which agents participate in each round of the tournament. The idea is not to do it automatically, but let the person running the tournament simply list the agent names for the next round in a file, or similar (see [Dennis' explanation](https://github.com/KWARC/Kalah-Framework/issues/6#issuecomment-562056674)).

**Comment by Jazzpirate** [comment]
If anyone wants to take care of this, I'm not opposed at all...

**Comment by katjabercic** [comment]
I believe the plan for this year is to figure out the specs, and then try to have it ready for next year, quite possibly with @tkw1536 's wizarding powers.

**Comment by Jazzpirate** [comment]
some thoughts:
1. adding submissions with something like a config file rather than coding should require some scala reflection, but should be doable in principle. Probably not even that much work, although cleaning up namespaces etc. will still be necessary.
2. Logging exceptions should be doable - I'd refrain from automatically disqualifying agents though. Sometimes, the exceptions are just a minor thing, so I prefer writing lots of forum posts informing people, so they can still fix those while the tournament is already running.
3. Separating out agents automatically seems... tricky. There's some "using my best judgment" involved, not just wrt where a noticable gap between scores is, but I'm usually also influenced there by what the resulting points would be. For example, if there's a clear gap between the best 8 and the ones below, I might choose a lower cutoff, because having the next (e.g.) three agents participate further might change who gets 9th and 10th place respectively, which makes a difference regarding how many bonus points they'll get...
point being: this is rather subtle ;)

**Comment by katjabercic** [comment]
Agreed on everything - with separating out I meant from the code :)
